File: python/ch1/ex115.py
import unittest
import math

def cube(x):
    return x * x * x


def p(x):
    return 3 * x - 4 * cube(x)


def sine(angle):
    if abs(angle) < 0.1:
        return angle
    else:
        return p(sine(angle / 3.0))



class MyTestCase(unittest.TestCase):
    def test1(self):
        self.assertEqual(sine(0), 0)

    def test2(self):
        print("Just printing.")
        print(sine(2))
        print(sine(math.pi / 6))


if __name__ == '__main__':
    unittest.main()

File: ruby/ch1/ex115.rb
require 'test/unit'

module Ex115
  def cube(x)
    x * x * x
  end

  def p(x)
    3 * x - 4 * cube(x)
  end

  def sine(angle)
    if angle.abs < 0.1
      angle
    else
      p(sine(angle / 3.0))
    end
  end
end

class MyTest < Test::Unit::TestCase
  include Ex115

  # Called before every test method runs. Can be used
  # to set up fixture information.
  def setup
    # Do nothing
  end

  # Called after every test method runs. Can be used to tear
  # down fixture information.

  def teardown
    # Do nothing
  end

  def test1
    assert(sine(0) == 0)
    assert(sine(1) > 0.84)
    assert(sine(2) < 0.91)
    assert(sine(3) < 0.15)
  end
end